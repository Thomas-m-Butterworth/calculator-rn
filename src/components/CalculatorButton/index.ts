export * from "./ButtonText";
export * from "./CalculatorButton";
