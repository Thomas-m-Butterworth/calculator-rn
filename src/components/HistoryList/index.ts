export * from "./Expression";
export * from "./EmptyHistoryList";
export * from "./HistoryList";
