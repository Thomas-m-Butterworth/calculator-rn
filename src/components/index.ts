export * from "./Calculator";
export * from "./CalculatorButton";
export * from "./CalculatorDisplay";
export * from "./HistoryList";
export * from "./ui";
