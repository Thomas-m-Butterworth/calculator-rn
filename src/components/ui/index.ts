export * from "./PageContainer";
export * from "./DisplayText";
