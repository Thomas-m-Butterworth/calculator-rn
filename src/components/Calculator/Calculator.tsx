import React from "react";
import { StyleSheet, useColorScheme, View } from "react-native";
import { Colors } from "@/constants/Colors";
import { useButtonLayout } from "@/src/hooks";
import { useCalculatorStore } from "@store/calculatorStore";
import { CalculatorButton } from "../CalculatorButton";
import { CalculatorDisplay } from "../CalculatorDisplay";

export const Calculator = () => {
  const {
    displayValue,
    appendDigit,
    removeDigit,
    setOperation,
    calculate,
    calculationHistory,
    clear,
    setPercentOperation,
  } = useCalculatorStore();

  const buttonLayout = useButtonLayout(
    appendDigit,
    removeDigit,
    setOperation,
    calculate,
    clear,
    setPercentOperation
  );

  const colorScheme = useColorScheme();
  const backgroundColor = Colors[colorScheme ?? "light"].background;

  return (
    <View style={[styles.container, { backgroundColor }]}>
      <CalculatorDisplay
        displayValue={displayValue}
        calculationHistory={calculationHistory}
      />
      <View style={styles.buttonGrid}>
        {buttonLayout.map((row, rowIndex) => (
          <View key={rowIndex} style={styles.buttonRow}>
            {row.map((buttonProps, buttonIndex) => (
              <CalculatorButton key={buttonIndex} {...buttonProps} />
            ))}
          </View>
        ))}
      </View>
    </View>
  );
};

const styles = StyleSheet.create({
  container: {
    justifyContent: "center",
    flex: 1,
    padding: 12,
  },
  buttonGrid: {
    justifyContent: "center",
    flex: 1,
    gap: 12,
  },
  buttonRow: {
    flexDirection: "row",
    gap: 12,
  },
});
