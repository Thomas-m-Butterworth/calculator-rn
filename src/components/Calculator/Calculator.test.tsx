import React from "react";
import { render } from "@testing-library/react-native";
import { Calculator } from "@/src/components/Calculator";

jest.mock("expo-font", () => ({
  loadAsync: jest.fn(() => Promise.resolve()),
  isLoaded: jest.fn(() => true),
}));

jest.mock("@/src/store/calculatorStore", () => ({
  useCalculatorStore: jest.fn(() => ({
    displayValue: "123",
    calculationHistory: ["12", "+", "10"],
    appendDigit: jest.fn(),
    removeDigit: jest.fn(),
    setOperation: jest.fn(),
    calculate: jest.fn(),
    clear: jest.fn(),
    setPercentOperation: jest.fn(),
  })),
}));

describe("Calculator", () => {
  it("should render as expected", () => {
    const { toJSON } = render(<Calculator />);
    expect(toJSON()).toMatchSnapshot();
  });
});
