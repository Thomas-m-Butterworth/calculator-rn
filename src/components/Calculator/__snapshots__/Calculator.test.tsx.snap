// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Calculator should render as expected 1`] = `
<View
  style={
    [
      {
        "flex": 1,
        "justifyContent": "center",
        "padding": 12,
      },
      {
        "backgroundColor": "#E3F2FD",
      },
    ]
  }
  testID="calculator"
>
  <View
    style={
      [
        {
          "alignItems": "flex-end",
          "borderRadius": 16,
          "height": 170,
          "justifyContent": "flex-end",
          "marginTop": 18,
          "padding": 16,
        },
        {
          "backgroundColor": "#FFFFFF",
        },
      ]
    }
    testID="calculatorDisplay"
  >
    <Text
      adjustsFontSizeToFit={true}
      numberOfLines={1}
      style={
        [
          {
            "fontFamily": "SpaceMono",
            "fontSize": 48,
            "fontWeight": "600",
          },
          {
            "color": "#263238",
          },
          undefined,
        ]
      }
      testID="calculatorDisplayText"
    >
      123
    </Text>
    <Text
      adjustsFontSizeToFit={true}
      numberOfLines={2}
      style={
        [
          {
            "fontFamily": "SpaceMono",
            "fontSize": 20,
            "letterSpacing": -0.5,
            "position": "absolute",
            "right": 16,
            "textAlign": "right",
            "top": 16,
          },
          {
            "color": "#666666",
          },
          undefined,
        ]
      }
      testID="calculatorHistoryText"
    >
      12 + 10
    </Text>
  </View>
  <View
    style={
      {
        "flex": 1,
        "gap": 12,
        "justifyContent": "center",
      }
    }
    testID="buttonGrid"
  >
    <View
      style={
        {
          "flexDirection": "row",
          "gap": 12,
        }
      }
    >
      <View
        accessibilityState={
          {
            "busy": undefined,
            "checked": undefined,
            "disabled": undefined,
            "expanded": undefined,
            "selected": undefined,
          }
        }
        accessibilityValue={
          {
            "max": undefined,
            "min": undefined,
            "now": undefined,
            "text": undefined,
          }
        }
        accessible={true}
        collapsable={false}
        focusable={true}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          {
            "alignItems": "center",
            "aspectRatio": 2,
            "backgroundColor": "#FF6F00",
            "borderRadius": 16,
            "elevation": 3,
            "flex": 2,
            "justifyContent": "center",
            "opacity": 1,
            "shadowColor": "#000",
            "shadowOffset": {
              "height": 2,
              "width": 0,
            },
            "shadowOpacity": 0.1,
            "shadowRadius": 4,
          }
        }
        testID="calc-button-0"
      >
        <Text
          style={
            [
              {
                "fontSize": 24,
                "fontWeight": "600",
              },
              {
                "color": "#000",
              },
              {
                "color": "#FFFFFF",
              },
            ]
          }
        >
          AC
        </Text>
      </View>
      <View
        accessibilityState={
          {
            "busy": undefined,
            "checked": undefined,
            "disabled": undefined,
            "expanded": undefined,
            "selected": undefined,
          }
        }
        accessibilityValue={
          {
            "max": undefined,
            "min": undefined,
            "now": undefined,
            "text": undefined,
          }
        }
        accessible={true}
        collapsable={false}
        focusable={true}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          {
            "alignItems": "center",
            "aspectRatio": 1,
            "backgroundColor": "#FF6F00",
            "borderRadius": 16,
            "elevation": 3,
            "flex": 1,
            "justifyContent": "center",
            "opacity": 1,
            "shadowColor": "#000",
            "shadowOffset": {
              "height": 2,
              "width": 0,
            },
            "shadowOpacity": 0.1,
            "shadowRadius": 4,
          }
        }
        testID="calc-button-1"
      >
        <Text
          style={
            [
              {
                "fontSize": 24,
                "fontWeight": "600",
              },
              {
                "color": "#000",
              },
              {
                "color": "#FFFFFF",
              },
            ]
          }
        >
          %
        </Text>
      </View>
      <View
        accessibilityState={
          {
            "busy": undefined,
            "checked": undefined,
            "disabled": undefined,
            "expanded": undefined,
            "selected": undefined,
          }
        }
        accessibilityValue={
          {
            "max": undefined,
            "min": undefined,
            "now": undefined,
            "text": undefined,
          }
        }
        accessible={true}
        collapsable={false}
        focusable={true}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          {
            "alignItems": "center",
            "aspectRatio": 1,
            "backgroundColor": "#FF6F00",
            "borderRadius": 16,
            "elevation": 3,
            "flex": 1,
            "justifyContent": "center",
            "opacity": 1,
            "shadowColor": "#000",
            "shadowOffset": {
              "height": 2,
              "width": 0,
            },
            "shadowOpacity": 0.1,
            "shadowRadius": 4,
          }
        }
        testID="calc-button-2"
      >
        <Text
          style={
            [
              {
                "fontSize": 24,
                "fontWeight": "600",
              },
              {
                "color": "#000",
              },
              {
                "color": "#FFFFFF",
              },
            ]
          }
        >
          รท
        </Text>
      </View>
    </View>
    <View
      style={
        {
          "flexDirection": "row",
          "gap": 12,
        }
      }
    >
      <View
        accessibilityState={
          {
            "busy": undefined,
            "checked": undefined,
            "disabled": undefined,
            "expanded": undefined,
            "selected": undefined,
          }
        }
        accessibilityValue={
          {
            "max": undefined,
            "min": undefined,
            "now": undefined,
            "text": undefined,
          }
        }
        accessible={true}
        collapsable={false}
        focusable={true}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          {
            "alignItems": "center",
            "aspectRatio": 1,
            "backgroundColor": "#FFFFFF",
            "borderRadius": 16,
            "elevation": 3,
            "flex": 1,
            "justifyContent": "center",
            "opacity": 1,
            "shadowColor": "#000",
            "shadowOffset": {
              "height": 2,
              "width": 0,
            },
            "shadowOpacity": 0.1,
            "shadowRadius": 4,
          }
        }
        testID="calc-button-0"
      >
        <Text
          style={
            [
              {
                "fontSize": 24,
                "fontWeight": "600",
              },
              {
                "color": "#000",
              },
              false,
            ]
          }
        >
          7
        </Text>
      </View>
      <View
        accessibilityState={
          {
            "busy": undefined,
            "checked": undefined,
            "disabled": undefined,
            "expanded": undefined,
            "selected": undefined,
          }
        }
        accessibilityValue={
          {
            "max": undefined,
            "min": undefined,
            "now": undefined,
            "text": undefined,
          }
        }
        accessible={true}
        collapsable={false}
        focusable={true}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          {
            "alignItems": "center",
            "aspectRatio": 1,
            "backgroundColor": "#FFFFFF",
            "borderRadius": 16,
            "elevation": 3,
            "flex": 1,
            "justifyContent": "center",
            "opacity": 1,
            "shadowColor": "#000",
            "shadowOffset": {
              "height": 2,
              "width": 0,
            },
            "shadowOpacity": 0.1,
            "shadowRadius": 4,
          }
        }
        testID="calc-button-1"
      >
        <Text
          style={
            [
              {
                "fontSize": 24,
                "fontWeight": "600",
              },
              {
                "color": "#000",
              },
              false,
            ]
          }
        >
          8
        </Text>
      </View>
      <View
        accessibilityState={
          {
            "busy": undefined,
            "checked": undefined,
            "disabled": undefined,
            "expanded": undefined,
            "selected": undefined,
          }
        }
        accessibilityValue={
          {
            "max": undefined,
            "min": undefined,
            "now": undefined,
            "text": undefined,
          }
        }
        accessible={true}
        collapsable={false}
        focusable={true}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          {
            "alignItems": "center",
            "aspectRatio": 1,
            "backgroundColor": "#FFFFFF",
            "borderRadius": 16,
            "elevation": 3,
            "flex": 1,
            "justifyContent": "center",
            "opacity": 1,
            "shadowColor": "#000",
            "shadowOffset": {
              "height": 2,
              "width": 0,
            },
            "shadowOpacity": 0.1,
            "shadowRadius": 4,
          }
        }
        testID="calc-button-2"
      >
        <Text
          style={
            [
              {
                "fontSize": 24,
                "fontWeight": "600",
              },
              {
                "color": "#000",
              },
              false,
            ]
          }
        >
          9
        </Text>
      </View>
      <View
        accessibilityState={
          {
            "busy": undefined,
            "checked": undefined,
            "disabled": undefined,
            "expanded": undefined,
            "selected": undefined,
          }
        }
        accessibilityValue={
          {
            "max": undefined,
            "min": undefined,
            "now": undefined,
            "text": undefined,
          }
        }
        accessible={true}
        collapsable={false}
        focusable={true}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          {
            "alignItems": "center",
            "aspectRatio": 1,
            "backgroundColor": "#FF6F00",
            "borderRadius": 16,
            "elevation": 3,
            "flex": 1,
            "justifyContent": "center",
            "opacity": 1,
            "shadowColor": "#000",
            "shadowOffset": {
              "height": 2,
              "width": 0,
            },
            "shadowOpacity": 0.1,
            "shadowRadius": 4,
          }
        }
        testID="calc-button-3"
      >
        <Text
          style={
            [
              {
                "fontSize": 24,
                "fontWeight": "600",
              },
              {
                "color": "#000",
              },
              {
                "color": "#FFFFFF",
              },
            ]
          }
        >
          ร
        </Text>
      </View>
    </View>
    <View
      style={
        {
          "flexDirection": "row",
          "gap": 12,
        }
      }
    >
      <View
        accessibilityState={
          {
            "busy": undefined,
            "checked": undefined,
            "disabled": undefined,
            "expanded": undefined,
            "selected": undefined,
          }
        }
        accessibilityValue={
          {
            "max": undefined,
            "min": undefined,
            "now": undefined,
            "text": undefined,
          }
        }
        accessible={true}
        collapsable={false}
        focusable={true}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          {
            "alignItems": "center",
            "aspectRatio": 1,
            "backgroundColor": "#FFFFFF",
            "borderRadius": 16,
            "elevation": 3,
            "flex": 1,
            "justifyContent": "center",
            "opacity": 1,
            "shadowColor": "#000",
            "shadowOffset": {
              "height": 2,
              "width": 0,
            },
            "shadowOpacity": 0.1,
            "shadowRadius": 4,
          }
        }
        testID="calc-button-0"
      >
        <Text
          style={
            [
              {
                "fontSize": 24,
                "fontWeight": "600",
              },
              {
                "color": "#000",
              },
              false,
            ]
          }
        >
          4
        </Text>
      </View>
      <View
        accessibilityState={
          {
            "busy": undefined,
            "checked": undefined,
            "disabled": undefined,
            "expanded": undefined,
            "selected": undefined,
          }
        }
        accessibilityValue={
          {
            "max": undefined,
            "min": undefined,
            "now": undefined,
            "text": undefined,
          }
        }
        accessible={true}
        collapsable={false}
        focusable={true}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          {
            "alignItems": "center",
            "aspectRatio": 1,
            "backgroundColor": "#FFFFFF",
            "borderRadius": 16,
            "elevation": 3,
            "flex": 1,
            "justifyContent": "center",
            "opacity": 1,
            "shadowColor": "#000",
            "shadowOffset": {
              "height": 2,
              "width": 0,
            },
            "shadowOpacity": 0.1,
            "shadowRadius": 4,
          }
        }
        testID="calc-button-1"
      >
        <Text
          style={
            [
              {
                "fontSize": 24,
                "fontWeight": "600",
              },
              {
                "color": "#000",
              },
              false,
            ]
          }
        >
          5
        </Text>
      </View>
      <View
        accessibilityState={
          {
            "busy": undefined,
            "checked": undefined,
            "disabled": undefined,
            "expanded": undefined,
            "selected": undefined,
          }
        }
        accessibilityValue={
          {
            "max": undefined,
            "min": undefined,
            "now": undefined,
            "text": undefined,
          }
        }
        accessible={true}
        collapsable={false}
        focusable={true}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          {
            "alignItems": "center",
            "aspectRatio": 1,
            "backgroundColor": "#FFFFFF",
            "borderRadius": 16,
            "elevation": 3,
            "flex": 1,
            "justifyContent": "center",
            "opacity": 1,
            "shadowColor": "#000",
            "shadowOffset": {
              "height": 2,
              "width": 0,
            },
            "shadowOpacity": 0.1,
            "shadowRadius": 4,
          }
        }
        testID="calc-button-2"
      >
        <Text
          style={
            [
              {
                "fontSize": 24,
                "fontWeight": "600",
              },
              {
                "color": "#000",
              },
              false,
            ]
          }
        >
          6
        </Text>
      </View>
      <View
        accessibilityState={
          {
            "busy": undefined,
            "checked": undefined,
            "disabled": undefined,
            "expanded": undefined,
            "selected": undefined,
          }
        }
        accessibilityValue={
          {
            "max": undefined,
            "min": undefined,
            "now": undefined,
            "text": undefined,
          }
        }
        accessible={true}
        collapsable={false}
        focusable={true}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          {
            "alignItems": "center",
            "aspectRatio": 1,
            "backgroundColor": "#FF6F00",
            "borderRadius": 16,
            "elevation": 3,
            "flex": 1,
            "justifyContent": "center",
            "opacity": 1,
            "shadowColor": "#000",
            "shadowOffset": {
              "height": 2,
              "width": 0,
            },
            "shadowOpacity": 0.1,
            "shadowRadius": 4,
          }
        }
        testID="calc-button-3"
      >
        <Text
          style={
            [
              {
                "fontSize": 24,
                "fontWeight": "600",
              },
              {
                "color": "#000",
              },
              {
                "color": "#FFFFFF",
              },
            ]
          }
        >
          -
        </Text>
      </View>
    </View>
    <View
      style={
        {
          "flexDirection": "row",
          "gap": 12,
        }
      }
    >
      <View
        accessibilityState={
          {
            "busy": undefined,
            "checked": undefined,
            "disabled": undefined,
            "expanded": undefined,
            "selected": undefined,
          }
        }
        accessibilityValue={
          {
            "max": undefined,
            "min": undefined,
            "now": undefined,
            "text": undefined,
          }
        }
        accessible={true}
        collapsable={false}
        focusable={true}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          {
            "alignItems": "center",
            "aspectRatio": 1,
            "backgroundColor": "#FFFFFF",
            "borderRadius": 16,
            "elevation": 3,
            "flex": 1,
            "justifyContent": "center",
            "opacity": 1,
            "shadowColor": "#000",
            "shadowOffset": {
              "height": 2,
              "width": 0,
            },
            "shadowOpacity": 0.1,
            "shadowRadius": 4,
          }
        }
        testID="calc-button-0"
      >
        <Text
          style={
            [
              {
                "fontSize": 24,
                "fontWeight": "600",
              },
              {
                "color": "#000",
              },
              false,
            ]
          }
        >
          1
        </Text>
      </View>
      <View
        accessibilityState={
          {
            "busy": undefined,
            "checked": undefined,
            "disabled": undefined,
            "expanded": undefined,
            "selected": undefined,
          }
        }
        accessibilityValue={
          {
            "max": undefined,
            "min": undefined,
            "now": undefined,
            "text": undefined,
          }
        }
        accessible={true}
        collapsable={false}
        focusable={true}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          {
            "alignItems": "center",
            "aspectRatio": 1,
            "backgroundColor": "#FFFFFF",
            "borderRadius": 16,
            "elevation": 3,
            "flex": 1,
            "justifyContent": "center",
            "opacity": 1,
            "shadowColor": "#000",
            "shadowOffset": {
              "height": 2,
              "width": 0,
            },
            "shadowOpacity": 0.1,
            "shadowRadius": 4,
          }
        }
        testID="calc-button-1"
      >
        <Text
          style={
            [
              {
                "fontSize": 24,
                "fontWeight": "600",
              },
              {
                "color": "#000",
              },
              false,
            ]
          }
        >
          2
        </Text>
      </View>
      <View
        accessibilityState={
          {
            "busy": undefined,
            "checked": undefined,
            "disabled": undefined,
            "expanded": undefined,
            "selected": undefined,
          }
        }
        accessibilityValue={
          {
            "max": undefined,
            "min": undefined,
            "now": undefined,
            "text": undefined,
          }
        }
        accessible={true}
        collapsable={false}
        focusable={true}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          {
            "alignItems": "center",
            "aspectRatio": 1,
            "backgroundColor": "#FFFFFF",
            "borderRadius": 16,
            "elevation": 3,
            "flex": 1,
            "justifyContent": "center",
            "opacity": 1,
            "shadowColor": "#000",
            "shadowOffset": {
              "height": 2,
              "width": 0,
            },
            "shadowOpacity": 0.1,
            "shadowRadius": 4,
          }
        }
        testID="calc-button-2"
      >
        <Text
          style={
            [
              {
                "fontSize": 24,
                "fontWeight": "600",
              },
              {
                "color": "#000",
              },
              false,
            ]
          }
        >
          3
        </Text>
      </View>
      <View
        accessibilityState={
          {
            "busy": undefined,
            "checked": undefined,
            "disabled": undefined,
            "expanded": undefined,
            "selected": undefined,
          }
        }
        accessibilityValue={
          {
            "max": undefined,
            "min": undefined,
            "now": undefined,
            "text": undefined,
          }
        }
        accessible={true}
        collapsable={false}
        focusable={true}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          {
            "alignItems": "center",
            "aspectRatio": 1,
            "backgroundColor": "#FF6F00",
            "borderRadius": 16,
            "elevation": 3,
            "flex": 1,
            "justifyContent": "center",
            "opacity": 1,
            "shadowColor": "#000",
            "shadowOffset": {
              "height": 2,
              "width": 0,
            },
            "shadowOpacity": 0.1,
            "shadowRadius": 4,
          }
        }
        testID="calc-button-3"
      >
        <Text
          style={
            [
              {
                "fontSize": 24,
                "fontWeight": "600",
              },
              {
                "color": "#000",
              },
              {
                "color": "#FFFFFF",
              },
            ]
          }
        >
          +
        </Text>
      </View>
    </View>
    <View
      style={
        {
          "flexDirection": "row",
          "gap": 12,
        }
      }
    >
      <View
        accessibilityState={
          {
            "busy": undefined,
            "checked": undefined,
            "disabled": undefined,
            "expanded": undefined,
            "selected": undefined,
          }
        }
        accessibilityValue={
          {
            "max": undefined,
            "min": undefined,
            "now": undefined,
            "text": undefined,
          }
        }
        accessible={true}
        collapsable={false}
        focusable={true}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          {
            "alignItems": "center",
            "aspectRatio": 1,
            "backgroundColor": "#FFFFFF",
            "borderRadius": 16,
            "elevation": 3,
            "flex": 1,
            "justifyContent": "center",
            "opacity": 1,
            "shadowColor": "#000",
            "shadowOffset": {
              "height": 2,
              "width": 0,
            },
            "shadowOpacity": 0.1,
            "shadowRadius": 4,
          }
        }
        testID="calc-button-0"
      >
        <Text
          style={
            [
              {
                "fontSize": 24,
                "fontWeight": "600",
              },
              {
                "color": "#000",
              },
              false,
            ]
          }
        >
          0
        </Text>
      </View>
      <View
        accessibilityState={
          {
            "busy": undefined,
            "checked": undefined,
            "disabled": undefined,
            "expanded": undefined,
            "selected": undefined,
          }
        }
        accessibilityValue={
          {
            "max": undefined,
            "min": undefined,
            "now": undefined,
            "text": undefined,
          }
        }
        accessible={true}
        collapsable={false}
        focusable={true}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          {
            "alignItems": "center",
            "aspectRatio": 1,
            "backgroundColor": "#FFFFFF",
            "borderRadius": 16,
            "elevation": 3,
            "flex": 1,
            "justifyContent": "center",
            "opacity": 1,
            "shadowColor": "#000",
            "shadowOffset": {
              "height": 2,
              "width": 0,
            },
            "shadowOpacity": 0.1,
            "shadowRadius": 4,
          }
        }
        testID="calc-button-1"
      >
        <Text
          style={
            [
              {
                "fontSize": 24,
                "fontWeight": "600",
              },
              {
                "color": "#000",
              },
              false,
            ]
          }
        >
          .
        </Text>
      </View>
      <View
        accessibilityState={
          {
            "busy": undefined,
            "checked": undefined,
            "disabled": undefined,
            "expanded": undefined,
            "selected": undefined,
          }
        }
        accessibilityValue={
          {
            "max": undefined,
            "min": undefined,
            "now": undefined,
            "text": undefined,
          }
        }
        accessible={true}
        collapsable={false}
        focusable={true}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          {
            "alignItems": "center",
            "aspectRatio": 1,
            "backgroundColor": "#FFFFFF",
            "borderRadius": 16,
            "elevation": 3,
            "flex": 1,
            "justifyContent": "center",
            "opacity": 1,
            "shadowColor": "#000",
            "shadowOffset": {
              "height": 2,
              "width": 0,
            },
            "shadowOpacity": 0.1,
            "shadowRadius": 4,
          }
        }
        testID="calc-button-2"
      >
        <Text
          allowFontScaling={false}
          selectable={false}
          style={
            [
              {
                "color": "#37474F",
                "fontSize": 32,
              },
              undefined,
              {
                "fontFamily": "ionicons",
                "fontStyle": "normal",
                "fontWeight": "normal",
              },
              {},
            ]
          }
        >
          ๎ฉ
        </Text>
      </View>
      <View
        accessibilityState={
          {
            "busy": undefined,
            "checked": undefined,
            "disabled": undefined,
            "expanded": undefined,
            "selected": undefined,
          }
        }
        accessibilityValue={
          {
            "max": undefined,
            "min": undefined,
            "now": undefined,
            "text": undefined,
          }
        }
        accessible={true}
        collapsable={false}
        focusable={true}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          {
            "alignItems": "center",
            "aspectRatio": 1,
            "backgroundColor": "#FF6F00",
            "borderRadius": 16,
            "elevation": 3,
            "flex": 1,
            "justifyContent": "center",
            "opacity": 1,
            "shadowColor": "#000",
            "shadowOffset": {
              "height": 2,
              "width": 0,
            },
            "shadowOpacity": 0.1,
            "shadowRadius": 4,
          }
        }
        testID="calc-button-3"
      >
        <Text
          style={
            [
              {
                "fontSize": 24,
                "fontWeight": "600",
              },
              {
                "color": "#000",
              },
              {
                "color": "#FFFFFF",
              },
            ]
          }
        >
          =
        </Text>
      </View>
    </View>
  </View>
</View>
`;
